client/my-sites/checkout/controller.jsx

```diff
@@ -157,6 +162,9 @@ export function checkout( context, next ) {
 		if ( isJetpackCheckout ) {
 			return true;
 		}
+		if ( context.query.source === 'my-jetpack' ) {
```

Comment:
WDYT about creating constant for `'my-jetpack'`? I don't know - do we have some module with similar constants to put it there?